packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -51,15 +55,19 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
     error L2_TOO_LATE();
 
     /// @notice Initializes the TaikoL2 contract.
-    /// @param _addressManager Address of the {AddressManager} contract.
+    /// @param _signalService Address of the {ISignalService} contract.
+    /// @param _gasExcess The initial gasExcess.
     function init(
-        address _addressManager,
+        address _signalService,
         uint64 _gasExcess
     )
         external
         initializer
     {
-        EssentialContract._init(_addressManager);
+        Ownable2StepUpgradeable.__Ownable2Step_init();
+
+        if (_signalService == address(0)) revert L2_INVALID_PARAM();
+        signalService = _signalService;
```

Comment:
Below there's logic for both `if (block.number > 1)` and `if (block.number > 0) {`. Shouldn't we just force the function to be called in a single block number? That would make things more clear.

Reply:
block number can only be 0 (real L2 genesi) or 1 (in tests). 
I optimized the code to make it easier to understand.